managed/src/main/java/com/yugabyte/yw/controllers/CustomerController.java

```diff
@@ -237,40 +240,27 @@ public Result metrics(UUID customerUUID) {
         .map((universe -> universe.getUniverseDetails().nodePrefix)).collect(Collectors.joining("|"));
       filterJson.put(universeFilterLabel, String.join("|", universePrefixes));
     } else {
-      // Check if it is a kubernetes deployment.
+      // Check if it is a Kubernetes deployment.
       if (hasContainerMetric) {
+        final String nodePrefix = params.remove("nodePrefix");
         if (params.containsKey("nodeName")) {
-          // Get the correct namespace by appending the zone if it exists.
+          // We calculate the correct namespace by using the zone if it exists.
           String[] nodeWithZone = params.remove("nodeName").split("_");
           filterJson.put(nodeFilterLabel, nodeWithZone[0]);
+          // TODO(bhavin192): might need to account for multiple
```

Comment:
a lot of this confusion will disappear if we change this api to specify  a metrics filter using universeUUID or clusterUUID instead of using these node prefixes.